pkg/morph/client/notifications.go

```diff
@@ -120,22 +120,23 @@ func (c *Client) ReceiveNotaryRequests(txSigner util.Uint160) error {
 		return ErrConnectionLost
 	}
 
-	c.subs.Lock()
-	defer c.subs.Unlock()
+	c.subs.RLock()
 
-	if c.subs.subscribedToAllNotaryEvents {
+	if _, ok := c.subs.subscribedNotaryEvents[txSigner]; ok || c.subs.subscribedToAllNotaryEvents {
```

Comment:
no need to touch `subscribedNotaryEvents` if `subscribedToAllNotaryEvents`

Reply:
is it critical? do not want more lines for not really important things

Reply:
not critical if this were code from scratch, but it is not. Extra lines are not a problem if they have a reason